relayer/crates/starknet-integration-tests/src/tests/clear_packets.rs

```diff
@@ -642,17 +654,59 @@ fn test_query_unreceived_packets() -> Result<(), Error> {
         )
         .await?;
 
-        // assert_eq!(pending_packets_starknet, vec![Sequence::from(3)]);
-        // TODO: Currently, querying for unreceived packets using commitment sequences that were never sent
-        // will always return them as unreceived. This assertion must be updated once the correct commitment-
-        // querying logic has been implemented.
-        // assert_eq!(pending_packets_cosmos, vec![Sequence::from(1), Sequence::from(2), Sequence::from(3)]);
-        // assert_eq!(pending_acks_starknet, vec![Sequence::from(1)]);
-        // assert_eq!(pending_acks_cosmos, vec![Sequence::from(2), Sequence::from(3)]);
-
-        // TODO: Call packet clearing
-        // TODO: Assert there are no pending packets or pending acks
-        // TODO: Assert tokens have been correctly transferred
+        let eventual_cosmos_amount = &Amount::new(
+            balance_cosmos_a_step_0.quantity - (transfer_quantity * 2) + transfer_back_quantity,
+            denom_cosmos.clone(),
+        );
+
+        birelay.auto_bi_relay(Some(Duration::from_secs(10)), Some(Duration::from_secs(0))).await?;
+
+        cosmos_chain
+            .assert_eventual_amount(address_cosmos_a, eventual_cosmos_amount)
+            .await?;
+
+        let balance_starknet_b_step_2 = starknet_chain
+            .query_token_balance(&ics20_token_address, address_starknet_b)
+            .await?;
+
+        assert_eq!(
+            balance_starknet_b_step_0.quantity + (transfer_quantity * 2).into()
+                - transfer_back_quantity.into(),
+            balance_starknet_b_step_2.quantity
+        );
+
+        let cosmos_latest_height = cosmos_chain.query_chain_height().await?;
+
+        let maybe_cosmos_ack1 = <CosmosChain as CanQueryPacketAcknowledgement<StarknetChain>>::query_packet_acknowledgement(
+            cosmos_chain,
+            &cosmos_channel_id,
+                &IbcPortId::transfer(),
+                &Sequence::from(1),
+                &cosmos_latest_height,
+            )
+            .await;
+
+        let maybe_cosmos_ack2 = <CosmosChain as CanQueryPacketAcknowledgement<StarknetChain>>::query_packet_acknowledgement(
+            cosmos_chain,
+            &cosmos_channel_id,
+                &IbcPortId::transfer(),
+                &Sequence::from(2),
+                &cosmos_latest_height,
+            )
+            .await;
+
+        let maybe_cosmos_ack3 = <CosmosChain as CanQueryPacketAcknowledgement<StarknetChain>>::query_packet_acknowledgement(
+            cosmos_chain,
+            &cosmos_channel_id,
+                &IbcPortId::transfer(),
+                &Sequence::from(3),
+                &cosmos_latest_height,
+            )
+            .await;
```

Comment:
shouldn't these be `query_packet_commitment`?

Reply:
Fixed in 736ff05ef5d3b8b52d2495a4ba5625bf70252cf3